### DISCUSSION TITLE
Regex replace including right square bracket but is ignored?

### DISCUSSION THREAD

**Discussion by lzwaan** [vectordotdev/vector#17691] [Q&A]
I tried to replace the name "PC-TEST in string for several days without luck

Orginal text:
`"stations\":[[\"F4:B5:20:XX:YY:ZZ\",\"PC-TEST\"]]`

Via the following code, ending with a square bracket "]" (?P<b>.*?"\])
`replace(string!(.data.raw_data), r'(?P<a>ZZ(.*?"){2})([a-zA-Z0-9_-]+)(?P<b>.*?"\])', "$${a}ANONYMIZED$${b}")`

Preferred text:
`"stations\":[[\"f4:b5:20:XX:YY:ZZ\",\"ANONYMIZED\"]]`

Without the closing ] bracket it will work but then it replaces some other text as well, how can i use a square bracket in this search?
Alse tried to replace the square bracket in the replace function with [[:punct:]] which will work but also replaces other text character?

**Comment by StephenWakely** [comment]
What is the result you are getting. When I run your code in the repl I get the preferred text that you have indicated:

```coffee
$ d = "stations\":[[\"F4:B5:20:XX:YY:ZZ\",\"PC-TEST\"]]"
"stations\":[[\"F4:B5:20:XX:YY:ZZ\",\"PC-TEST\"]]"

$ replace(d, r'(?P<a>ZZ(.*?"){2})([a-zA-Z0-9_-]+)(?P<b>.*?"\])', "${a}ANONYMIZED${b}")
"stations\":[[\"F4:B5:20:XX:YY:ZZ\",\"ANONYMIZED\"]]"
```
